### DISCUSSION TITLE
v4.4.0 release draft

### DISCUSSION THREAD

**Discussion by luu-alex** [ethereum/web3.js#6728] [Announcements]
#### web3-eth

-   Catch `TransactionPollingTimeoutError` was added to send transaction events (#6623)

#### web3-eth-abi

-   Use `AbiError` instead of `Error` for errors at web3-eth-abi (#6641).

#### web3-eth-contract

-	Allow the `deploy` function to accept parameters, even when no ABI was provided to the `Contract`(#6635)

#### web3

-   Dependencies updated

#### web3-eth-abi

-   Fixed an issue with detecting Uint8Array (#6486)

#### web3-eth-accounts

-   Send Transaction config used to be ignored if the passed `common` did not have a `copy()` and the `chainId` was not provided (#6663)
-   Fixed an issue with detecting Uint8Array (#6486)

#### web3-eth-contract

-	Fix and error that happen when trying to get past events by calling `contract.getPastEvents` or `contract.events.allEvents()`, if there is no matching events. (#6647)
-	Fixed: The Contract is not using the context wallet passed if context was passed at constructor. (#6661)

#### web3-utils

-   Fixed an issue with detecting Uint8Array (#6486)

#### web3-validator

-   Fixed an issue with detecting Uint8Array (#6486)

<hr /><em>This discussion was created from the release <a href='https://github.com/web3/web3.js/releases/tag/v4.4.0'>v4.4.0 release draft</a>.</em>

**Comment by jessbennett** [comment]
> ## [4.4.0]
> ### Added
> #### web3-eth
> * Catch `TransactionPollingTimeoutError` was added to send transaction events ([Errors not caught in 4.x #6623](https://github.com/web3/web3.js/pull/6623))
> 
> ### Changed
> #### web3-eth-abi
> * Use `AbiError` instead of `Error` for errors at web3-eth-abi ([Use AbiError instead of Error for errors at web3-eth-abi #6641](https://github.com/web3/web3.js/pull/6641)).
> 
> #### web3-eth-contract
> * Allow the `deploy` function to accept parameters, even when no ABI was provided to the `Contract`([Allow the `deploy` function to accept parameters even when no ABI was provided to the Contract #6635](https://github.com/web3/web3.js/pull/6635))
> 
> #### web3
> * Dependencies updated
> 
> ### Fixed
> #### web3-eth-abi
> * Fixed an issue with detecting Uint8Array ([Fix an issue with detecting Uint8Array #6486](https://github.com/web3/web3.js/pull/6486))
> 
> #### web3-eth-accounts
> * Send Transaction config used to be ignored if the passed `common` did not have a `copy()` and the `chainId` was not provided ([Increase Unit test coverage for web3-eth above 90% #6663](https://github.com/web3/web3.js/pull/6663))
> * Fixed an issue with detecting Uint8Array ([Fix an issue with detecting Uint8Array #6486](https://github.com/web3/web3.js/pull/6486))
> 
> #### web3-eth-contract
> * Fix and error that happen when trying to get past events by calling `contract.getPastEvents` or `contract.events.allEvents()`, if there is no matching events. ([Fix and error when trying to get past events by calling `contract.getPastEvents` or `contract.events.allEvents()` if there is no matching events. #6647](https://github.com/web3/web3.js/pull/6647))
> * Fixed: The Contract is not using the context wallet passed if context was passed at constructor. ([fix: The Contract is not using the context wallet passed if context was passed at constructor #6661](https://github.com/web3/web3.js/pull/6661))
> 
> #### web3-utils
> * Fixed an issue with detecting Uint8Array ([Fix an issue with detecting Uint8Array #6486](https://github.com/web3/web3.js/pull/6486))
> 
> #### web3-validator
> * Fixed an issue with detecting Uint8Array ([Fix an issue with detecting Uint8Array #6486](https://github.com/web3/web3.js/pull/6486))
> 
> _This discussion was created from the release [v4.4.0 release draft](https://github.com/web3/web3.js/releases/tag/v4.4.0)._

☁️
